scipp-developer.yml

```diff
@@ -1,53 +0,0 @@
-# This is used for incremental CI builds and local development.
-# All dependencies should be installed here (i.e. the union of build, run, test and documentation build dependencies).
-# See https://scipp.github.io/developer/dependencies.html
-
-name: scipp-developer
-
-channels:
-  - conda-forge
-
-dependencies:
-  # Build
-  - ninja
-  - tbb-devel
-  - conan
-  - cmake
-  - cppcheck
-  - pip
-  - gxx_linux-64 11.1.*
-
-  # Run
-  - confuse
-  - ipykernel = 6.3.1  # issue with pythreejs 3d plotting
-  - ipympl
-  - ipywidgets
-  - matplotlib-base
-  - numpy>=1.20.0
-  - python
-  - pythreejs
-  - pyyaml
-  - tbb
-  - graphlib-backport  # only needed for py < 3.9
-
-  # Test
-  - h5py
-  - pytest
-  - python-graphviz
-  - xarray  # For testing xarray <-> scipp conversion
-  - pandas  # For testing pandas <-> scipp conversion
-  - scipy>=1.7.0
-
-  # Formatting & static analysis
-  - pre-commit
-  - yapf
-  - flake8
-  - pip:
-    - nb-clean
```

Comment:
These are not included in meta.yml. Is there a way to provide them in the dev env anyway?

Reply:
we can make a small env file and use the `--merge-with` argument of the `metatoenv.py` script?

Reply:
`developer-extra.yml`?

Reply:
Note that these are also missing from `scippneutron`. I should add them there also, if that is the solution we want to go with.

Reply:
Maybe. It would be good to use the same file for CI so that the pinned versions stay in sync. I'd say we think about this in a separate step. It shouldn't stop this PR.